code/controllers/ModelAsController.php

```diff
@@ -49,82 +49,79 @@ public function init() {
 	public function handleRequest(SS_HTTPRequest $request, DataModel $model) {
 		$this->setRequest($request);
 		$this->setDataModel($model);
-		
-		$this->pushCurrent();
 
-		// Create a response just in case init() decides to redirect
-		$this->response = new SS_HTTPResponse();
```

Comment:
It looks like the response was intentionally created *before* `init()` is called here. Have you checked there are no regressions if `init()` is relying on `$this->response`?